### ISSUE TITLE
Bugs with new Tokenising

### ISSUE THREAD

**Issue by littleStudent** [Krzysztof-Cieslak/vscode-elm#233]
## 1.
<img width="503" alt="screenshot 2018-07-31 11 20 08" src="https://user-images.githubusercontent.com/815520/43450771-c4238cf4-94b3-11e8-8089-8507eda9a474.png">

`appModel` seems to be treated like this:

`Model` ->
```
* support.other.module.elm
* support.other.module.name.elm
* source.elm
```

`app` ->
```
* source.elm
```

## 2.
<img width="309" alt="screenshot 2018-07-31 11 22 56" src="https://user-images.githubusercontent.com/815520/43450958-289a474a-94b4-11e8-9056-4da94bad255c.png">

I am not able to color functions like `split` anymore.
They are tagged with `source.elm`

## 3.
<img width="303" alt="screenshot 2018-07-31 11 24 57" src="https://user-images.githubusercontent.com/815520/43451064-5eda1ace-94b4-11e8-9e0b-d7dca592ce53.png">

The second type `LoggedIn_NoProfile` is treated differently?
Using anything `snake_case` will break the tokenising as well.
This happens for function names and types.

**Comment by JasonMFry** [comment]
Having same issue, notice `selectableCourse` is sometimes colored correctly, other times not.

![screen shot 2018-07-31 at 8 27 46 am](https://user-images.githubusercontent.com/166681/43459506-a866b2d8-949b-11e8-9611-994739504c66.png)

**Comment by Krzysztof-Cieslak** [comment]
CC: @mdgriffith

**Comment by mdgriffith** [comment]
I'm on it :)

I've fixed #1 and #3 locally.  #2 is because core functions are no longer given special treatment.  BUT I realized I forgot to add a tag for functions in general, so I can do that.

I also realized I forgot to cover `glsl` blocks, so I'm working on that.
